src/System.Private.CoreLib/src/System/Reflection/Emit/DynamicILGenerator.cs

```diff
@@ -875,6 +875,18 @@ internal class DynamicILInfo
         private int m_methodSignature;
         #endregion
 
+        #region Constructor
+        internal DynamicILInfo(object scope, DynamicMethod method, byte[] methodSignature)
```

Comment:
Why did you change the first argument type from DynamicScope to object? It adds unnecessary cast below.

Reply:
BTW: The DynamicScope argument looks useless. DynamicScope is internal detail of DynamicILInfo. You can just do `new DynamicScope` inside the constructor. It does not need to be passed in from outside.

Reply:
Yes, the original reason was because I didn't want to have to expose that type.  Best is to instantiate it in ctor as you suggested.